core/src/main/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlanner.java

```diff
@@ -161,8 +161,13 @@ public void init(InitParameters params) {
 
       CompactionExecutorId ceid;
 
-      Objects.requireNonNull(executorConfig.type,
-          "'type' is a required and must be 'internal' or 'external'");
+      // If not supplied, GSON will leave type null. Default to internal
+      if (null == executorConfig.type) {
+        executorConfig.type = "internal";
+      } else if (!("internal".equals(executorConfig.type)
+          || "external".equals(executorConfig.type))) {
```

Comment:
I've noticed that Eclipse has some settings that suggest expanding negation of compound boolean expressions. I like these suggestions, because I think it's easier for the human mind to interpret the compound of separate negations ("it's not a pig and it's not a dog") than the negation of a compound ("it's not the case that it's a pig or a dog").

So, I suggest:

```java
      } else if (!"internal".equals(executorConfig.type)
          && !"external".equals(executorConfig.type)) {
```

Or, maybe something with set inclusion instead:

```suggestion
      } else if (!Set.of("internal", "external").contains(executorConfig.type)) {
```

Or, just remove this check entirely, since the default case in the switch statement just below it already handles this.

Reply:
Good call on the removal, it is handled below that. I'm removing this.

Reply:
Removed in f271227